Brian Holliday<br/>
Professor Masuda<br/>
Mathematical Modeling I<br/>
20 December 2019<br/><br/>

## Project 3

In [1]:
load('linProg.sage')

#### Part A

In [9]:
v = ['x11','x12','x13','x14',
    'x21','x22','x23','x24',#variables
    'x31','x32','x33','x34']
v2 = ['Mill_1', 'Mill_2','Mill_3',
      'MI','NY','CA','GA']#constraits 
f = [430,550,680,700,
     510,590,890,685,
     395,425,910,450]
c = [105,160,85,70,100,105,75]
m = [1,1,1,-1,-1,-1,-1] 
M = [[1,1,1,1, 0,0,0,0, 0,0,0,0], #Mill_1
     [0,0,0,0, 1,1,1,1, 0,0,0,0], #Mill_2
     [0,0,0,0, 0,0,0,0, 1,1,1,1], #Mill_3
     [1,0,0,0, 1,0,0,0, 1,0,0,0], #Michigan
     [0,1,0,0, 0,1,0,0, 0,1,0,0], #New York
     [0,0,1,0, 0,0,1,0, 0,0,1,0], #California
     [0,0,0,1, 0,0,0,1, 0,0,0,1]] #Georgia
lp(v, v2, f, M, m, c, mx=False)

GLPK 4.63 - SENSITIVITY ANALYSIS REPORT                                                                         Page   1
 
 Problem:    
 Objective:  198200 (MINimum)
 
    No. Row name     St      Activity         Slack   Lower bound       Activity      Obj coef  Obj value at Limiting
                                           Marginal   Upper bound          range         range   break point variable
 ------ ------------ -- ------------- ------------- -------------  ------------- ------------- ------------- ------------
      1 Mill_1       NU     105.00000        .               -Inf      105.00000          -Inf  198200.00000 x11
                                          -80.00000     105.00000      175.00000      80.00000  192600.00000 x21
 
      2 Mill_2       BS     160.00000        .               -Inf      160.00000     -80.00000  185400.00000 Mill_1
                                             .          160.00000      160.00000          +Inf          +Inf
 
      3 Mill_3    

The mimimum cost for shipping is: $198,200

There are 105 truck loads from Textile Mill 1 to California<br/>
There are 70 truck loads from Textile Mill 2 to Michigan<br/>
There are 90 truck loads from Textile Mill 2 to New York<br/>
There are 10 truck loads from Textile Mill 3 to New York<br/>
There are 75 truck loads from Textile Mill 3 to Georgia

#### Part B
Shadow Prices

In [11]:

c = [106,160,85,70,100,105,75]
m = [1,1,1,-1,-1,-1,-1]
lp(v, v2, f, M, m, c, mx=False)

GLPK 4.63 - SENSITIVITY ANALYSIS REPORT                                                                         Page   1
 
 Problem:    
 Objective:  198120 (MINimum)
 
    No. Row name     St      Activity         Slack   Lower bound       Activity      Obj coef  Obj value at Limiting
                                           Marginal   Upper bound          range         range   break point variable
 ------ ------------ -- ------------- ------------- -------------  ------------- ------------- ------------- ------------
      1 Mill_1       NU     106.00000        .               -Inf      105.00000          -Inf  198200.00000 Mill_2
                                          -80.00000     106.00000      175.00000      80.00000  192600.00000 x21
 
      2 Mill_2       BS     159.00000       1.00000          -Inf      160.00000     -80.00000  185400.00000 Mill_1
                                             .          160.00000      159.00000          +Inf          +Inf
 
      3 Mill_3 

In [17]:
198200 - 198120 
#an increase by 1 in the output of Mill 1 causes a 80 dollar decrese in the transportation

80

In [15]:
c = [105,161,85,70,100,105,75]
m = [1,1,1,-1,-1,-1,-1]
lp(v, v2, f, M, m, c, mx=False)

GLPK 4.63 - SENSITIVITY ANALYSIS REPORT                                                                         Page   1
 
 Problem:    
 Objective:  198200 (MINimum)
 
    No. Row name     St      Activity         Slack   Lower bound       Activity      Obj coef  Obj value at Limiting
                                           Marginal   Upper bound          range         range   break point variable
 ------ ------------ -- ------------- ------------- -------------  ------------- ------------- ------------- ------------
      1 Mill_1       NU     105.00000        .               -Inf      105.00000          -Inf  198200.00000 x11
                                          -80.00000     105.00000      175.00000      80.00000  192600.00000 x21
 
      2 Mill_2       BS     160.00000       1.00000          -Inf      160.00000     -80.00000  185400.00000 Mill_1
                                             .          161.00000      160.00000          +Inf          +Inf
 
      3 Mill_3    

An increase of one in the output of Mill 2 leaves us with the same transportation cost

In [16]:
c = [105,160,86,70,100,105,75]
m = [1,1,1,-1,-1,-1,-1]
lp(v, v2, f, M, m, c, mx=False)

GLPK 4.63 - SENSITIVITY ANALYSIS REPORT                                                                         Page   1
 
 Problem:    
 Objective:  198035 (MINimum)
 
    No. Row name     St      Activity         Slack   Lower bound       Activity      Obj coef  Obj value at Limiting
                                           Marginal   Upper bound          range         range   break point variable
 ------ ------------ -- ------------- ------------- -------------  ------------- ------------- ------------- ------------
      1 Mill_1       NU     105.00000        .               -Inf      105.00000          -Inf  198035.00000 x11
                                          -80.00000     105.00000      175.00000      80.00000  192435.00000 x21
 
      2 Mill_2       BS     159.00000       1.00000          -Inf      159.00000     -80.00000  185315.00000 Mill_1
                                             .          160.00000      159.00000          +Inf          +Inf
 
      3 Mill_3    

In [18]:
198200 - 198035

165

An increase of one in the output of Mill 3 causes a decrease in the transportation cost of 165

#### Part C

Imagine if the shipping cost we cut in half for all of the routes

In [35]:
v = ['x11','x12','x13','x14',
    'x21','x22','x23','x24',#variables
    'x31','x32','x33','x34']
v2 = ['Mill_1', 'Mill_2','Mill_3',
      'MI','NY','CA','GA']#constraits 
f = [430/2,550/2,680/2,700/2,
     510/2,590/2,890/2,685/2,
     395/2,425/2,910/2,450/2]
c = [105,160,85,70,100,105,75]
m = [1,1,1,-1,-1,-1,-1]
M = [[1,1,1,1, 0,0,0,0, 0,0,0,0], #Mill_1
     [0,0,0,0, 1,1,1,1, 0,0,0,0], #Mill_2
     [0,0,0,0, 0,0,0,0, 1,1,1,1], #Mill_3
     [1,0,0,0, 1,0,0,0, 1,0,0,0], #Michigan
     [0,1,0,0, 0,1,0,0, 0,1,0,0], #New York
     [0,0,1,0, 0,0,1,0, 0,0,1,0], #California
     [0,0,0,1, 0,0,0,1, 0,0,0,1]] #Georgia
lpInt(v, v2, f, M, m, c, mx=False) #Integer Programming Method

Optimal (min) value:  99100.0

Variable      Value
----------  -------
x11               0
x12               0
x13             105
x14               0
x21              70
x22              90
x23               0
x24               0
x31               0
x32              10
x33               0
x34              75

Constraint      Slack
------------  -------
Mill_1              0
Mill_2              0
Mill_3              0
MI                  0
NY                  0
CA                  0
GA                  0


We have the same allocation for the shipments as Part A, so we have to see which shipments cost the most:<br/>

In [36]:
105*(680/2) #x13

35700

In [37]:
70 * (510/2) #x21

17850

In [38]:
90 * (590/2) #x22

26550

In [32]:
10 * 425 #x32

4250

In [33]:
75 * 450 #x34

33750

We could see that much of the shipping cost is being taken by x13, x21, and x22. So these routes will be given to the fuel efficient trucks<br/>
New Cost:118100<br/>
Savings: 80100

#### Part D

4 trucks

In [39]:
v = ['x11','x12','x13','x14',
    'x21','x22','x23','x24',#variables
    'x31','x32','x33','x34']
v2 = ['Mill_1', 'Mill_2','Mill_3',
      'MI','NY','CA','GA']#constraits 
f = [430,550,680/2,700,
     510/2,590/2,890,685,
     395,425,910,450/2]
c = [105,160,85,70,100,105,75]
m = [1,1,1,-1,-1,-1,-1]
M = [[1,1,1,1, 0,0,0,0, 0,0,0,0], #Mill_1
     [0,0,0,0, 1,1,1,1, 0,0,0,0], #Mill_2
     [0,0,0,0, 0,0,0,0, 1,1,1,1], #Mill_3
     [1,0,0,0, 1,0,0,0, 1,0,0,0], #Michigan
     [0,1,0,0, 0,1,0,0, 0,1,0,0], #New York
     [0,0,1,0, 0,0,1,0, 0,0,1,0], #California
     [0,0,0,1, 0,0,0,1, 0,0,0,1]] #Georgia
lp(v, v2, f, M, m, c, mx=False)

GLPK 4.63 - SENSITIVITY ANALYSIS REPORT                                                                         Page   1
 
 Problem:    
 Objective:  101225 (MINimum)
 
    No. Row name     St      Activity         Slack   Lower bound       Activity      Obj coef  Obj value at Limiting
                                           Marginal   Upper bound          range         range   break point variable
 ------ ------------ -- ------------- ------------- -------------  ------------- ------------- ------------- ------------
      1 Mill_1       NU     105.00000        .               -Inf      105.00000          -Inf  101225.00000 Mill_3
                                         -570.00000     105.00000      105.00000     570.00000  101225.00000 x33
 
      2 Mill_2       NU     160.00000        .               -Inf      160.00000          -Inf  101225.00000 Mill_3
                                         -130.00000     160.00000      170.00000     130.00000   99925.00000 x32
 
      3 Mil

We see that having 4 half price truck gives us a cost of 101225
Total Cost: 149224

In [40]:
101224 + (4)*12000

149224

In [41]:
v = ['x11','x12','x13','x14',
    'x21','x22','x23','x24',#variables
    'x31','x32','x33','x34']
v2 = ['Mill_1', 'Mill_2','Mill_3',
      'MI','NY','CA','GA']#constraits 
f = [430,550,680/2,700,
     510/2,590/2,890,685,
     395,425/2,910,450/2]
c = [105,160,85,70,100,105,75]
m = [1,1,1,-1,-1,-1,-1]
M = [[1,1,1,1, 0,0,0,0, 0,0,0,0], #Mill_1
     [0,0,0,0, 1,1,1,1, 0,0,0,0], #Mill_2
     [0,0,0,0, 0,0,0,0, 1,1,1,1], #Mill_3
     [1,0,0,0, 1,0,0,0, 1,0,0,0], #Michigan
     [0,1,0,0, 0,1,0,0, 0,1,0,0], #New York
     [0,0,1,0, 0,0,1,0, 0,0,1,0], #California
     [0,0,0,1, 0,0,0,1, 0,0,0,1]] #Georgia
lp(v, v2, f, M, m, c, mx=False)

GLPK 4.63 - SENSITIVITY ANALYSIS REPORT                                                                         Page   1
 
 Problem:    
 Objective:  99100 (MINimum)
 
    No. Row name     St      Activity         Slack   Lower bound       Activity      Obj coef  Obj value at Limiting
                                           Marginal   Upper bound          range         range   break point variable
 ------ ------------ -- ------------- ------------- -------------  ------------- ------------- ------------- ------------
      1 Mill_1       NU     105.00000        .               -Inf      105.00000          -Inf   99100.00000 Mill_2
                                         -550.00000     105.00000      105.00000     550.00000   99100.00000 x23
 
      2 Mill_2       BS     160.00000        .               -Inf      170.00000     -82.50000   85900.00000 Mill_3
                                             .          160.00000      160.00000          +Inf          +Inf
 
      3 Mill_3  

We see that having 5 half price trucks gives us a cost of 99100<br/>
New Cost: 

In [43]:
99100 + (5)*12000

159100

In [44]:
159100 - 149224

9876

We could see that there is almost a 10,000 savings by having 4 trucks. Therefore 4 fuel efficient trucks is more cost effective

#### Part E

In [4]:
v = ['x11','x12','x13','x14',
    'x21','x22','x23','x24',#variables
    'x31','x32','x33','x34']
v2 = ['Mill_1', 'Mill_2','Mill_3',
      'MI','NY','CA','GA']#constraits 
f = [430,550,680/2,700,
     510,590,890/2,685,
     395,425,910/2,450]
c = [105,160,85,70,100,105,75]
m = [1,1,1,-1,-1,-1,-1]
M = [[1,1,1,1, 0,0,0,0, 0,0,0,0], #Mill_1
     [0,0,0,0, 1,1,1,1, 0,0,0,0], #Mill_2
     [0,0,0,0, 0,0,0,0, 1,1,1,1], #Mill_3
     [1,0,0,0, 1,0,0,0, 1,0,0,0], #Michigan
     [0,1,0,0, 0,1,0,0, 0,1,0,0], #New York
     [0,0,1,0, 0,0,1,0, 0,0,1,0], #California
     [0,0,0,1, 0,0,0,1, 0,0,0,1]] #Georgia
lp(v, v2, f, M, m, c, mx=False)

GLPK 4.63 - SENSITIVITY ANALYSIS REPORT                                                                         Page   1
 
 Problem:    
 Objective:  162500 (MINimum)
 
    No. Row name     St      Activity         Slack   Lower bound       Activity      Obj coef  Obj value at Limiting
                                           Marginal   Upper bound          range         range   break point variable
 ------ ------------ -- ------------- ------------- -------------  ------------- ------------- ------------- ------------
      1 Mill_1       NU     105.00000        .               -Inf      105.00000          -Inf  162500.00000 x11
                                          -80.00000     105.00000      175.00000      80.00000  156900.00000 x21
 
      2 Mill_2       BS     160.00000        .               -Inf      160.00000     -80.00000  149700.00000 Mill_1
                                             .          160.00000      160.00000          +Inf          +Inf
 
      3 Mill_3    

In [5]:
198200 - 162500

35700

If you must use energy efficient trucks on the California route, you could save $35700

### Problem 16

#### Part A

In [48]:
v = ['Model A', 'Model B', 'Model C']#Computer Models
v2 = ['Hours', 'Monitor A', 'Monitor B', 'Total Monitors']
f = [(1250 - 850),(1400-1250),(2500-1500)]#Sales Price and Cost of each model
c = [20000,5000,7500,10000]#Contraints
m = [1,1,1,1]
M = [[1,1.25,1.75],#Hours
    [1,1,0],       #Monitor A
    [0,0,1],       #Monitor B
    [1,1,1]]       #Total Monitors 
lp(v, v2, f, M, m, c, mx=True)#Max

GLPK 4.63 - SENSITIVITY ANALYSIS REPORT                                                                         Page   1
 
 Problem:    
 Objective:  8500000 (MAXimum)
 
    No. Row name     St      Activity         Slack   Lower bound       Activity      Obj coef  Obj value at Limiting
                                           Marginal   Upper bound          range         range   break point variable
 ------ ------------ -- ------------- ------------- -------------  ------------- ------------- ------------- ------------
      1 Hours        BS   15625.00000    4375.00000          -Inf    13125.00000    -400.00000      2.25e+06 Total Monitors
                                             .        20000.00000    16250.00000    1000.00000    2.4125e+07 Model B
 
      2 Monitor A    BS    2500.00000    2500.00000          -Inf         .         -400.00000       7.5e+06 Total Monitors
                                             .         5000.00000    10000.00000     600.00000         1e

From the report there are 2500 of Model A and 7500 of Model C

#### Part B

In [7]:
v = ['Model A', 'Model B', 'Model C']
v2 = ['Hours', 'Monitor A', 'Monitor B', 'Total Monitors']
f = [(1250 - 850),(1400-1250),(2500-1500)]
c = [20001,5000,7500,10000]#increase hours by 1 for shadow price
m = [1,1,1,1]
M = [[1,1.25,1.75],#Hours
    [1,1,0],       #Monitor A
    [0,0,1],       #Monitor B
    [1,1,1]]       #Total Monitors 
lp(v, v2, f, M, m, c, mx=True)

GLPK 4.63 - SENSITIVITY ANALYSIS REPORT                                                                         Page   1
 
 Problem:    
 Objective:  8500000 (MAXimum)
 
    No. Row name     St      Activity         Slack   Lower bound       Activity      Obj coef  Obj value at Limiting
                                           Marginal   Upper bound          range         range   break point variable
 ------ ------------ -- ------------- ------------- -------------  ------------- ------------- ------------- ------------
      1 Hours        BS   15625.00000    4376.00000          -Inf    13125.00000    -400.00000      2.25e+06 Total Monitors
                                             .        20001.00000    16250.00000    1000.00000    2.4125e+07 Model B
 
      2 Monitor A    BS    2500.00000    2500.00000          -Inf         .         -400.00000       7.5e+06 Total Monitors
                                             .         5000.00000    10000.00000     600.00000         1e

We see that the Hours constraint is non binding and has no effect on our profit

In [8]:
v = ['Model A', 'Model B', 'Model C']
v2 = ['Hours', 'Monitor A', 'Monitor B', 'Total Monitors']
f = [(1250 - 850),(1400-1250),(2500-1500)]
c = [20000,5001,7500,10000]#increase Monitor A by 1 for shadow price
m = [1,1,1,1]
M = [[1,1.25,1.75],#Hours
    [1,1,0],       #Monitor A
    [0,0,1],       #Monitor B
    [1,1,1]]       #Total Monitors 
lp(v, v2, f, M, m, c, mx=True)

GLPK 4.63 - SENSITIVITY ANALYSIS REPORT                                                                         Page   1
 
 Problem:    
 Objective:  8500000 (MAXimum)
 
    No. Row name     St      Activity         Slack   Lower bound       Activity      Obj coef  Obj value at Limiting
                                           Marginal   Upper bound          range         range   break point variable
 ------ ------------ -- ------------- ------------- -------------  ------------- ------------- ------------- ------------
      1 Hours        BS   15625.00000    4375.00000          -Inf    13125.00000    -400.00000      2.25e+06 Total Monitors
                                             .        20000.00000    16250.00000    1000.00000    2.4125e+07 Model B
 
      2 Monitor A    BS    2500.00000    2501.00000          -Inf         .         -400.00000       7.5e+06 Total Monitors
                                             .         5001.00000    10000.00000     600.00000         1e

We see that the Monitor A constraint is non binding and has no effect on our profit

In [9]:
v = ['Model A', 'Model B', 'Model C']
v2 = ['Hours', 'Monitor A', 'Monitor B', 'Total Monitors']
f = [(1250 - 850),(1400-1250),(2500-1500)]
c = [20000,5000,7501,10000]#increase Monitor B by 1 for shadow price
m = [1,1,1,1]
M = [[1,1.25,1.75],#Hours
    [1,1,0],       #Monitor A
    [0,0,1],       #Monitor B
    [1,1,1]]       #Total Monitors 
lp(v, v2, f, M, m, c, mx=True)

GLPK 4.63 - SENSITIVITY ANALYSIS REPORT                                                                         Page   1
 
 Problem:    
 Objective:  8500600 (MAXimum)
 
    No. Row name     St      Activity         Slack   Lower bound       Activity      Obj coef  Obj value at Limiting
                                           Marginal   Upper bound          range         range   break point variable
 ------ ------------ -- ------------- ------------- -------------  ------------- ------------- ------------- ------------
      1 Hours        BS   15625.75000    4374.25000          -Inf    13126.75000    -400.00000    2.2503e+06 Total Monitors
                                             .        20000.00000    16250.50000    1000.00000   2.41264e+07 Model B
 
      2 Monitor A    BS    2499.00000    2501.00000          -Inf         .         -400.00000     7.501e+06 Total Monitors
                                             .         5000.00000    10000.00000     600.00000         1e

In [10]:
8500600 - 8500000

600

Our shadow price for the Monitor B constraint is $600

In [12]:
v = ['Model A', 'Model B', 'Model C']
v2 = ['Hours', 'Monitor A', 'Monitor B', 'Total Monitors']
f = [(1250 - 850),(1400-1250),(2500-1500)]
c = [20000,5000,7500,10001]#increase Total Monitors by 1 for shadow price
m = [1,1,1,1]
M = [[1,1.25,1.75],#Hours
    [1,1,0],       #Monitor A
    [0,0,1],       #Monitor B
    [1,1,1]]       #Total Monitors 
lp(v, v2, f, M, m, c, mx=True)

GLPK 4.63 - SENSITIVITY ANALYSIS REPORT                                                                         Page   1
 
 Problem:    
 Objective:  8500400 (MAXimum)
 
    No. Row name     St      Activity         Slack   Lower bound       Activity      Obj coef  Obj value at Limiting
                                           Marginal   Upper bound          range         range   break point variable
 ------ ------------ -- ------------- ------------- -------------  ------------- ------------- ------------- ------------
      1 Hours        BS   15626.00000    4374.00000          -Inf    13125.00000    -400.00000      2.25e+06 Total Monitors
                                             .        20000.00000    16251.25000    1000.00000   2.41264e+07 Model B
 
      2 Monitor A    BS    2501.00000    2499.00000          -Inf         .         -400.00000       7.5e+06 Total Monitors
                                             .         5000.00000    10001.00000     600.00000    1.0001e

In [15]:
8500400 - 8500000

400

Our shadow price for the Total Monitors constraint is $600

##### Part C

Change sales price of Model C from 2500 to 2199

In [18]:
v = ['Model A', 'Model B', 'Model C']
v2 = ['Hours', 'Monitor A', 'Monitor B', 'Total Monitors']
f = [(1250 - 850),(1400-1250),(2199-1500)]#Change price of Model C from 2500 to 2199
c = [20000,5000,7500,10000]#Constraints
m = [1,1,1,1]
M = [[1,1.25,1.75],#Hours
    [1,1,0],       #Monitor A
    [0,0,1],       #Monitor B
    [1,1,1]]       #Total Monitors 
lp(v, v2, f, M, m, c, mx=True)

GLPK 4.63 - SENSITIVITY ANALYSIS REPORT                                                                         Page   1
 
 Problem:    
 Objective:  6242500 (MAXimum)
 
    No. Row name     St      Activity         Slack   Lower bound       Activity      Obj coef  Obj value at Limiting
                                           Marginal   Upper bound          range         range   break point variable
 ------ ------------ -- ------------- ------------- -------------  ------------- ------------- ------------- ------------
      1 Hours        BS   15625.00000    4375.00000          -Inf    13750.00000    -398.66667   13333.33333 Monitor B
                                             .        20000.00000    16250.00000    1000.00000   2.18675e+07 Model B
 
      2 Monitor A    BS    2500.00000    2500.00000          -Inf         .         -400.00000    5.2425e+06 Total Monitors
                                             .         5000.00000    10000.00000     299.00000      6.99e+06 M

In [2]:
8500000 - 6242500

2257500

Change in results from Part A:<br/>

It does not change in the breakdown in what Model are sold, but our profit is down $2,257,500

In [19]:
v = ['Model A', 'Model B', 'Model C']
v2 = ['Hours', 'Monitor A', 'Monitor B', 'Total Monitors']
f = [(1250 - 850),(1400-1250),(2199-1500)]
c = [20000,5000,7501,10000]#Increase Monitor B by 1
m = [1,1,1,1]
M = [[1,1.25,1.75],#Hours
    [1,1,0],       #Monitor A
    [0,0,1],       #Monitor B
    [1,1,1]]       #Total Monitors 
lp(v, v2, f, M, m, c, mx=True)

GLPK 4.63 - SENSITIVITY ANALYSIS REPORT                                                                         Page   1
 
 Problem:    
 Objective:  6242799 (MAXimum)
 
    No. Row name     St      Activity         Slack   Lower bound       Activity      Obj coef  Obj value at Limiting
                                           Marginal   Upper bound          range         range   break point variable
 ------ ------------ -- ------------- ------------- -------------  ------------- ------------- ------------- ------------
      1 Hours        BS   15625.75000    4374.25000          -Inf    13750.00000    -398.66667   13333.33333 Monitor B
                                             .        20000.00000    16250.50000    1000.00000   2.18685e+07 Model B
 
      2 Monitor A    BS    2499.00000    2501.00000          -Inf         .         -400.00000    5.2432e+06 Total Monitors
                                             .         5000.00000    10000.00000     299.00000      6.99e+06 M

In [21]:
6242799 - 6242500

299

In [22]:
v = ['Model A', 'Model B', 'Model C']
v2 = ['Hours', 'Monitor A', 'Monitor B', 'Total Monitors']
f = [(1250 - 850),(1400-1250),(2199-1500)]
c = [20000,5000,7500,10001]#Increase Total Monitors by 1
m = [1,1,1,1]
M = [[1,1.25,1.75],#Hours
    [1,1,0],       #Monitor A
    [0,0,1],       #Monitor B
    [1,1,1]]       #Total Monitors 
lp(v, v2, f, M, m, c, mx=True)

GLPK 4.63 - SENSITIVITY ANALYSIS REPORT                                                                         Page   1
 
 Problem:    
 Objective:  6242900 (MAXimum)
 
    No. Row name     St      Activity         Slack   Lower bound       Activity      Obj coef  Obj value at Limiting
                                           Marginal   Upper bound          range         range   break point variable
 ------ ------------ -- ------------- ------------- -------------  ------------- ------------- ------------- ------------
      1 Hours        BS   15626.00000    4374.00000          -Inf    13751.75000    -398.66667   13334.66667 Monitor B
                                             .        20000.00000    16251.25000    1000.00000   2.18689e+07 Model B
 
      2 Monitor A    BS    2501.00000    2499.00000          -Inf         .         -400.00000    5.2425e+06 Total Monitors
                                             .         5000.00000    10001.00000     299.00000    6.9907e+06 M

In [23]:
6242900 - 6242500

400

Change in results from Part B:<br/>
Shadow price for Monitor A goes from 600 to 299<br/>
Shadow price for Total Monitors remains the same

#### Part D

In [25]:
v = ['Model A', 'Model B', 'Model C','Model D']
v2 = ['Hours', 'Monitor A', 'Monitor B', 'Total Monitors']
f = [(1250 - 850),(1400-1250),(2500-1500),(1895 - 1250)]
c = [20000,5000,7500,10000]#Increase Total Monitors by 1
m = [1,1,1,1]
M = [[1,1.25,1.75,1.5],#Hours
    [1,1,0,0],       #Monitor A
    [0,0,1,1],       #Monitor B
    [1,1,1,1]]       #Total Monitors 
lp(v, v2, f, M, m, c, mx=True)

GLPK 4.63 - SENSITIVITY ANALYSIS REPORT                                                                         Page   1
 
 Problem:    
 Objective:  8500000 (MAXimum)
 
    No. Row name     St      Activity         Slack   Lower bound       Activity      Obj coef  Obj value at Limiting
                                           Marginal   Upper bound          range         range   break point variable
 ------ ------------ -- ------------- ------------- -------------  ------------- ------------- ------------- ------------
      1 Hours        BS   15625.00000    4375.00000          -Inf    13125.00000    -400.00000      2.25e+06 Total Monitors
                                             .        20000.00000    16250.00000    1000.00000    2.4125e+07 Model B
 
      2 Monitor A    BS    2500.00000    2500.00000          -Inf         .         -400.00000       7.5e+06 Total Monitors
                                             .         5000.00000    10000.00000     600.00000         1e

Adding Model D to our model changes nothing about our previous model.